packages/pyright-internal/src/languageService/definitionProvider.ts

```diff
@@ -207,6 +219,65 @@ export class DefinitionProvider {
         }
     }
 
+    private static _tryAddTypedDictDefinitions(
+        node: StringNode,
+        evaluator: TypeEvaluator,
+        definitions: DocumentRange[],
+        sourceMapper: SourceMapper
+    ) {
+        const type = evaluator.getExpectedType(node)?.type;
+        if (type === undefined) {
+            return;
+        }
+
+        doForEachSubtype(type, (subtype) => {
+            if (isClassInstance(subtype) && ClassType.isTypedDictClass(subtype)) {
+                const entry = subtype.details.typedDictEntries?.get(node.value);
+                if (entry !== undefined) {
+                    DefinitionProvider._resolveDeclarations(
+                        DefinitionProvider._resolveField(subtype, node.value)?.getDeclarations(),
+                        evaluator,
+                        definitions,
+                        sourceMapper
+                    );
+                }
+            }
+        });
+    }
+
+    private static _resolveField(type: ClassType, name: string, checkedTypes?: Set<TypeSourceId>): Symbol | undefined {
+        // resolve a field from the symbol table for the given class and all it's base classes
```

Comment:
Minor nits: Comments should generally be complete sentences starting with capital letter and ending in period. Also, "it's" should be "its".